dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -290,37 +292,48 @@ private CachedVanityUrl searchLiveCachedVanityUrlBySiteAndLanguage(final String
      * @param user to use in the ES search
      * @return A list of VanityURLs
      */
-    private List<VanityUrl> searchAndPopulate(final String luceneQuery, final User user) {
-
-        final ImmutableList.Builder<VanityUrl> results = new ImmutableList.Builder();
-        final PriorityQueue<VanityUrl> vanityUrls;
-        List<VanityUrl> vanityUrlsToReturn = Collections.emptyList();
-
-        try {
-            final List<Contentlet> contentResults = contentletAPI
-                    .search(luceneQuery, 0, 0, StringPool.BLANK, user, false);
-
-            //Verify if we have something to process
-            if (null == contentResults || contentResults.isEmpty()) {
-                return results.build();
-            }
-
-            vanityUrls = new PriorityQueue<>(contentResults.size(),
-                    (vanity1, vanity2) -> vanity1.getOrder() - vanity2.getOrder());
-
-            contentResults.stream().forEach((Contentlet con) -> {
-                VanityUrl vanityUrl = getVanityUrlFromContentlet(con);
-                addToVanityURLCache(vanityUrl);
-                vanityUrls.offer(vanityUrl);
-            });
-
-            vanityUrlsToReturn = results.addAll(vanityUrls).build();
-        } catch (DotDataException | DotSecurityException e) {
-            Logger.error(this, "Error searching for active Vanity URLs [" + luceneQuery + "]", e);
-        }
-
-        return vanityUrlsToReturn;
-    }
+	private List<VanityUrl> searchAndPopulate(final String luceneQuery, final User user) {
+
+		final ImmutableList.Builder<VanityUrl> results = new ImmutableList.Builder();
+		final PriorityQueue<VanityUrl> vanityUrls;
+		List<VanityUrl> vanityUrlsToReturn = Collections.emptyList();
+
+		try {
+			final List<Contentlet> contentResults = contentletAPI.search(luceneQuery, 0, 0, StringPool.BLANK, user,
+					false);
+
+			// Verify if we have something to process
+			if (null == contentResults || contentResults.isEmpty()) {
+				return results.build();
+			}
+
+			vanityUrls = new PriorityQueue<>(contentResults.size(),
+					(vanity1, vanity2) -> vanity1.getOrder() - vanity2.getOrder());
+
+			contentResults.stream().forEach((Contentlet con) -> {
+				VanityUrl vanityUrl = getVanityUrlFromContentlet(con);
+				addToVanityURLCache(vanityUrl);
+				vanityUrls.offer(vanityUrl);
+			});
+
+			vanityUrlsToReturn = results.addAll(vanityUrls).build();
+		} catch (Exception e) {
+			if (e instanceof IndexMissingException || e.getCause() instanceof IndexMissingException) {
+				/*
+				 * We catch this exception in order to avoid to stop the
+				 * initialization of dotCMS if for some reason at this point we
+				 * don't have indexes.
+				 */
+				Logger.error(this, "Error when initializing Vanity URLs, no index found ", e);
+			} else if (e instanceof DotDataException || e instanceof DotSecurityException) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)